rust/tools/authorization-logic/raksha_examples/VisionPipelineAuthLogic

```diff
@@ -0,0 +1,274 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+
+"EndUser" says {
+    ownsTag("raw_video_tag").
+    ownsTag("user_selection_tag").
+    ownsTag("product_id_tag").
+      
+    // These lines say the end user trusts (i.e., delegates to) the TPPR to
+    // faithfully say when a data transformation from raw camera video to the 
+    // name of a product has taken place, but only when the user has selected 
+    // an object.
+    //
+    // It might look like this ideally:
+    //
+    // "ThirdPartyPrivacyReviewer" canSay becomes(pathX, "raw_video_tag",
+    //              "product_id_tag") :-
+    //     objectSelected("EndUser", pathX),
+    //     isAccessPath(pathX).
+    //
+    // For now it has to be done like this:
+    "ThirdPartyPrivacyReviewer" canSay declassifies(pathX, "raw_video_tag") :-
+        hasTag(pathX, "product_id_tag"),   // raw_video_tag is only removed if 
+        objectSelected("EndUser", pathX),  // product_id_tag is added
+        isAccessPath(pathX).
+
+    "ThirdPartyPrivacyReviewer" canSay
+        objectSelected("EndUser", "identified_product").
+
+    // Delegates TPPR the ability to apply tags anywhere
+    "ThirdPartyPrivacyReviewer" canSay hasTag(x, y) :- 
+        isAccessPath(x), isTag(y).
+
+    // Delegate the right to claim that particles are "Local" to TPPR.
+    // (We can think of the principal "LocalDevice" as a group)
+    "ThirdPartyPrivacyReviewer" canSay principalX canActAs "LocalDevice" :-
+        isPrincipal(principalX).
+
+    "LocalDevice" may("do_pure_computation", "raw_video_tag").
```

Comment:
This is not handled as a distinct language feature. This is just a normal predicate at the moment. The language feature is avoided because we can probably get away with not having it while we are in a MVP phase.